src/util/signature_utils.js

```diff
@@ -372,9 +372,34 @@ function authorize_request_account_by_token(token, requesting_account, is_secret
 }
 
 
+// In case the object access_keys in requesting_account has the is_active property
+// and is_active is false we will not continue to work on the request and immediately throw an error
+// access_key_id_to_find is a string (not SensitiveString)
+// inside requesting_account the access keys ids are SensitiveStrings
+function check_access_key_status(access_key_id_to_find, requesting_account) {
+    const access_keys = requesting_account.access_keys;
+    const index = _.findIndex(access_keys, item => item.access_key.unwrap() === access_key_id_to_find);
```

Comment:
using find only in this flow seems missing because the other functions like `authorize_request_account_by_token` are just looking at key at index 0. shouldn't we add a function to look up an access key pair from the account and use it instead in the other flows?

Reply:
this case might be related to the rpc auth flow.